File: golang/hashmap/contains_duplicate.go
package main

func containsNearbyDuplicate(nums []int, k int) bool {
	numIndices := make(map[int]int)

	for i, num := range nums {
		if j, found := numIndices[num]; found && i-j <= k {
			return true
		}

		numIndices[num] = i
	}

	return false
}

File: java/src/main/java/com/sanedge/interview/hashmap/ContainsDuplicate.java
package com.sanedge.interview.hashmap;

import java.util.HashMap;
import java.util.Map;

public class ContainsDuplicate {
  public boolean containsNearbyDuplicate(int[] nums, int k) {
    Map<Integer, Integer> numsIndices = new HashMap<>();

    for (int i = 0; i < nums.length; i++) {
      int num = nums[i];

      if (numsIndices.containsKey(num) && i - numsIndices.get(num) <= k) {
        return true;
      }

      numsIndices.put(num, i);
    }

    return false;
  }
}
